src/org/moeaframework/core/variable/StepRealVariable.java

```diff
@@ -0,0 +1,181 @@
+/* Copyright 2009-2018 David Hadka
+ *
+ * This file is part of the MOEA Framework.
+ *
+ * The MOEA Framework is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or (at your
+ * option) any later version.
+ *
+ * The MOEA Framework is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public
+ * License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with the MOEA Framework.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.moeaframework.core.variable;
+
+import org.apache.commons.lang3.builder.EqualsBuilder;
+import org.apache.commons.lang3.builder.HashCodeBuilder;
+import org.moeaframework.core.PRNG;
+import org.moeaframework.core.Variable;
+
+import java.text.MessageFormat;
+
+/**
+ * Decision variable for doubleegers encoded as a binary string.  Note that if
+ * {@code upperBound-lowerBound} is not a power of 2, then some values will
+ * occur more frequently after a variation operator.
+ */
+public class StepRealVariable implements Variable {
```

Comment:
Can this extend from RealVariable?  This way, it can be used as a drop-in replacement and you can re-use some of the existing code in RealVariable.

Reply:
You are right. I am not familiar with JAVA, so I may take some time to modify it.